packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPluginCreateCameraTests.swift

```diff
@@ -0,0 +1,124 @@
+  private func createSutAndMocks() -> (
+    CameraPlugin, MockFLTCameraPermissionManager, MockCaptureSession
+  ) {
+    let mockPermissionManager = MockFLTCameraPermissionManager()
+    let mockCaptureSession = MockCaptureSession()
+
+    let cameraPlugin = CameraPlugin(
+      registry: MockFlutterTextureRegistry(),
+      messenger: MockFlutterBinaryMessenger(),
+      globalAPI: MockGlobalEventApi(),
+      deviceDiscoverer: MockCameraDeviceDiscoverer(),
+      permissionManager: mockPermissionManager,
+      deviceFactory: { _ in MockCaptureDevice() },
+      captureSessionFactory: { mockCaptureSession },
+      captureDeviceInputFactory: MockCaptureDeviceInputFactory()
+    )
+
+    return (cameraPlugin, mockPermissionManager, mockCaptureSession)
+  }
+
+  func testCreateCamera_requestsOnlyCameraPermissionWithAudioDisabled() {
+    let (cameraPlugin, mockPermissionManager, _) = createSutAndMocks()
+    let expectation = expectation(description: "Initialization completed")
+
+    var requestCameraPermissionCalled = false
+    mockPermissionManager.requestCameraPermissionStub = { completion in
+      requestCameraPermissionCalled = true
+      completion?(nil)
+    }
+    var requestAudioPermissionCalled = false
+    mockPermissionManager.requestAudioPermissionStub = { completion in
+      requestAudioPermissionCalled = true
+      completion?(nil)
+    }
+
+    cameraPlugin.createCamera(
+      withName: "camera_name",
+      settings: FCPPlatformMediaSettings.make(
+        with: .medium,
+        framesPerSecond: nil,
+        videoBitrate: nil,
+        audioBitrate: nil,
+        enableAudio: false)
+    ) { result, error in
+      expectation.fulfill()
+    }
+
+    waitForExpectations(timeout: 30, handler: nil)
+
+    XCTAssertTrue(requestCameraPermissionCalled)
+    XCTAssertFalse(requestAudioPermissionCalled)
+  }
+
+  func testCreateCamera_requestsCameraAndAudioPermissionWithAudioEnabled() {
+    let (cameraPlugin, mockPermissionManager, _) = createSutAndMocks()
+    let expectation = expectation(description: "Initialization completed")
+
+    var requestCameraPermissionCalled = false
+    mockPermissionManager.requestCameraPermissionStub = { completion in
+      requestCameraPermissionCalled = true
+      completion?(nil)
+    }
+    var requestAudioPermissionCalled = false
+    mockPermissionManager.requestAudioPermissionStub = { completion in
+      requestAudioPermissionCalled = true
+      completion?(nil)
+    }
+
+    cameraPlugin.createCamera(
+      withName: "camera_name",
+      settings: FCPPlatformMediaSettings.make(
+        with: .medium,
+        framesPerSecond: nil,
+        videoBitrate: nil,
+        audioBitrate: nil,
+        enableAudio: true)
+    ) { result, error in
+      expectation.fulfill()
+    }
+
+    waitForExpectations(timeout: 30, handler: nil)
+
+    XCTAssertTrue(requestCameraPermissionCalled)
+    XCTAssertTrue(requestAudioPermissionCalled)
+  }
+
+  func testCreateCamera_assignsCamera() {
```

Comment:
what does `assignsCamera` mean?

Reply:
We expect to have an instance of FLTCam assigned to the `camera` field after `createCamera` is executed

Reply:
That's an odd thing to test. A test should test its behavior, not the implementation details.

Reply:
I agree but I don't see a better way to test this. The ultimate goal of this method is to set up the FLTCam instance on which other methods depend. So the options are:
1. Verify the internal state (as it is currently)
2. Inject a camera factory and verify that it has been called (doesn't test quite everything)
3. Test this with other methods depending on FLTCam by using it in test setup (reduces the isolation between tests)
4. Extract this state outside of this class (possible but IMO overkill)

If you see a better way to test this or prefer one of the other approaches let me know

Reply:
Will other tests cover this part of code?

Reply:
No unless we switch to option (3) and actually depend on this method in other tests

Reply:
Can you call it `testCreateCamera_createsFLTCamSuccessfully()`